pkg/cloudprovider/provider/azure/provider.go

```diff
@@ -0,0 +1,615 @@
+	rawCfg := rawConfig{}
+	err = json.Unmarshal(pconfig.CloudProviderSpec.Raw, &rawCfg)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	c := config{}
+	c.SubscriptionID, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.SubscriptionID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"subscriptionID\" field, error = %v", err)
+	}
+
+	c.TenantID, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.TenantID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"tenantID\" field, error = %v", err)
+	}
+
+	c.ClientID, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.ClientID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"clientID\" field, error = %v", err)
+	}
+
+	c.ClientSecret, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.ClientSecret)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"clientSecret\" field, error = %v", err)
+	}
+
+	c.ResourceGroup, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.ResourceGroup)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"resourceGroup\" field, error = %v", err)
+	}
+
+	c.Location, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.Location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"location\" field, error = %v", err)
+	}
+
+	c.VMSize, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.VMSize)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"vmSize\" field, error = %v", err)
+	}
+
+	c.VNetName, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.VNetName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"vnetName\" field, error = %v", err)
+	}
+
+	c.SubnetName, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.SubnetName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"subnetName\" field, error = %v", err)
+	}
+
+	c.RouteTableName, err = p.configVarResolver.GetConfigVarStringValue(rawCfg.RouteTableName)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get the value of \"routeTableName\" field, error = %v", err)
+	}
+
+	return &c, &pconfig, nil
+}
+
+func getVMIPAddresses(ctx context.Context, c *config, vm *compute.VirtualMachine) ([]string, error) {
+	var ipAddresses []string
+
+	if vm.VirtualMachineProperties == nil {
+		return nil, fmt.Errorf("machine is missing properties")
+	}
+
+	if vm.VirtualMachineProperties.NetworkProfile == nil {
+		return nil, fmt.Errorf("machine has no network profile")
+	}
+
+	if vm.NetworkProfile.NetworkInterfaces == nil {
+		return nil, fmt.Errorf("machine has no network interfaces data")
+	}
+
+	for n, iface := range *vm.NetworkProfile.NetworkInterfaces {
+		if iface.ID == nil {
+			return nil, fmt.Errorf("interface %d has no ID", n)
+		}
+
+		splitIfaceID := strings.Split(*iface.ID, "/")
+		ifaceName := splitIfaceID[len(splitIfaceID)-1]
```

Comment:
Could you check the length of `splitIfaceID` before accessing a calculated index ;) will panic if the length is zero

Reply:
fixed

Reply:
:+1: